BUG: NumPy build failure on Alpine Linux s390x

### Describe the issue:

Failure to build on Alpine Linux s390x where `HWCAP_S390_VX` is not defined.
There's a patch downstream which won't work on glibc<2.33: https://gitlab.alpinelinux.org/alpine/aports/-/blob/v3.21.0/community/py3-numpy/s390x-hwcap.patch?ref_type=tags

### Reproduce the code example:

```python
in a musllinux_1_2_s390x container:
apk add openblas-dev
pip wheel -w $(pwd) --no-binary=numpy numpy==2.1.1
```


### Error message:

```shell
cc -Inumpy/core/_umath_tests.cpython-312-s390x-linux-musl.so.p -Inumpy/core -I../../numpy/core -I../../numpy/core/src/multiarray -I../../numpy/core/src/npymath -Inumpy/core/include -I../../numpy/core/include -I../../numpy/core/src/common -I/opt/_internal/cpython-3.12.8/include/python3.12 -I/tmp/pip-wheel-6fwvkqrl/numpy_c87a1d8e27ec475bbcf6ad8f75a59ba4/.mesonpy-91dpnx77/build/meson_cpu -fvisibility=hidden -fdiagnostics-color=always -DNDEBUG -Wall -Winvalid-pch -std=c99 -O3 -fno-strict-aliasing -fPIC -DNPY_INTERNAL_BUILD -DHAVE_NPY_CONFIG_H -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE=1 -D_LARGEFILE64_SOURCE=1 -MD -MQ numpy/core/_umath_tests.cpython-312-s390x-linux-musl.so.p/src_common_npy_cpu_features.c.o -MF numpy/core/_umath_tests.cpython-312-s390x-linux-musl.so.p/src_common_npy_cpu_features.c.o.d -o numpy/core/_umath_tests.cpython-312-s390x-linux-musl.so.p/src_common_npy_cpu_features.c.o -c ../../numpy/core/src/common/npy_cpu_features.c
      ../../numpy/core/src/common/npy_cpu_features.c: In function 'npy__cpu_init_features':
      ../../numpy/core/src/common/npy_cpu_features.c:640:18: error: 'HWCAP_S390_VX' undeclared (first use in this function); did you mean 'HWCAP_S390_VXE'?
        640 |     if ((hwcap & HWCAP_S390_VX) == 0) {
            |                  ^~~~~~~~~~~~~
            |                  HWCAP_S390_VXE

```
```


### Python and NumPy Versions:

Python 3.12
NumPy 2.1.1

### Runtime Environment:

_No response_

### Context for the issue:

_No response_

--- a/numpy/_core/src/common/npy_cpu_features.c
+++ b/numpy/_core/src/common/npy_cpu_features.c
@@ -632,11 +632,12 @@ npy__cpu_init_features(void)
 
 #include <sys/auxv.h>
 
-#ifndef HWCAP_S390_VX
-    #define HWCAP_S390_VX 2048
+/* kernel HWCAP names, available in musl, not available in glibc<2.33: https://sourceware.org/bugzilla/show_bug.cgi?id=25971 */
+#ifndef HWCAP_S390_VXRS
+    #define HWCAP_S390_VXRS 2048
 #endif
-#ifndef HWCAP_S390_VXE
-    #define HWCAP_S390_VXE 8192
+#ifndef HWCAP_S390_VXRS_EXT
+    #define HWCAP_S390_VXRS_EXT 8192
 #endif
 #ifndef HWCAP_S390_VXRS_EXT2
     #define HWCAP_S390_VXRS_EXT2 32768
@@ -648,7 +649,7 @@ npy__cpu_init_features(void)
     memset(npy__cpu_have, 0, sizeof(npy__cpu_have[0]) * NPY_CPU_FEATURE_MAX);
 
     unsigned int hwcap = getauxval(AT_HWCAP);
-    if ((hwcap & HWCAP_S390_VX) == 0) {
+    if ((hwcap & HWCAP_S390_VXRS) == 0) {
         return;
     }
 
@@ -659,7 +660,7 @@ npy__cpu_init_features(void)
        return;
     }
 
-    npy__cpu_have[NPY_CPU_FEATURE_VXE] = (hwcap & HWCAP_S390_VXE) != 0;
+    npy__cpu_have[NPY_CPU_FEATURE_VXE] = (hwcap & HWCAP_S390_VXRS_EXT) != 0;
 
     npy__cpu_have[NPY_CPU_FEATURE_VX]  = 1;
 }

--- a/numpy/_core/src/common/npy_cpu_features.c
+++ b/numpy/_core/src/common/npy_cpu_features.c
@@ -631,10 +631,13 @@ npy__cpu_init_features(void)
 #elif defined(__s390x__)
 
 #include <sys/auxv.h>
+
+#ifndef HWCAP_S390_VX
+    #define HWCAP_S390_VX 2048
+#endif
 #ifndef HWCAP_S390_VXE
     #define HWCAP_S390_VXE 8192
 #endif
-
 #ifndef HWCAP_S390_VXRS_EXT2
     #define HWCAP_S390_VXRS_EXT2 32768
 #endif

